### ISSUE TITLE
skipExportGlyphs regression in 2.18.0

### ISSUE THREAD

**Issue by kontur** [googlefonts/ufo2ft#432]
Following up to the release of 2.18.0 and the solution to not include `skipExportGlyphs` in the feature writer I am getting KeyErrors for the non-exporting glyphs.

```
  File "/Users/johannes/Packages/ufo2ft/Lib/ufo2ft/featureWriters/baseFeatureWriter.py", line 110, in write
    self.setContext(font, feaFile, compiler=compiler)
  File "/Users/johannes/Packages/ufo2ft/Lib/ufo2ft/featureWriters/kernFeatureWriter.py", line 211, in setContext
    ctx.kerning = self.getKerningData(font, feaFile, self.getOrderedGlyphSet())
  File "/Users/johannes/Packages/ufo2ft/Lib/ufo2ft/featureWriters/baseFeatureWriter.py", line 161, in getOrderedGlyphSet
    return OrderedDict((gn, glyphSet[gn]) for gn in glyphOrder)
  File "/Users/johannes/Packages/ufo2ft/Lib/ufo2ft/featureWriters/baseFeatureWriter.py", line 161, in <genexpr>
    return OrderedDict((gn, glyphSet[gn]) for gn in glyphOrder)
KeyError: '_part.markGaf'
```

If I understand correctly, [this code here](https://github.com/googlefonts/ufo2ft/blob/c44ca7275482956cec2c04ac666224bba23a2637/Lib/ufo2ft/featureWriters/baseFeatureWriter.py#L160) tries to create a new dict that includes everything in the `glyphOrder`.

I am not sure if this is caused by the way I am extracting the UFOs from .glyphs sources via glyphsLib or if this applicable to all UFOs. Either way, the UFO's `glyphs/contents.plist` does contain the non-exporting glyphs, so even though `glyphSet` is filtered by the skipExportGlyphs attribute, the following OrderedDict creation does try access the `glyphOrder` index of the non-exporting glyphs that have been pruned from `glyphSet`, and thus raises the error.

**Comment by anthrotype** [comment]
Thanks for the report and the PR. I have a similar fix coming, stay tuned

**Comment by anthrotype** [comment]
released v2.18.1 with the fix => https://pypi.org/project/ufo2ft/2.18.1

**Comment by kontur** [comment]
Thanks for the quick action. Ah, I see, there's a more convenient way already in the method signature! Good to include a test 👍
